src/_pytest/outcomes.py

```diff
@@ -77,132 +73,118 @@ def __init__(
         super().__init__(msg)
 
 
-# We need a callable protocol to add attributes, for discussion see
-# https://github.com/python/mypy/issues/2087.
+class XFailed(Failed):
+    """Raised from an explicit call to pytest.xfail()."""
 
-_F = TypeVar("_F", bound=Callable[..., object])
-_ET = TypeVar("_ET", bound=type[BaseException])
 
+class _Exit:
+    Exception: type[Exit] = Exit
```

Comment:
Usually we omit an explicit annotation when it is inferred.

Also, here I think a `ClassVar` annotation would be appropriate. So `Exception: ClassVar = Exit`.

Reply:
Good idea, added `ClassVar` -- see below comment re: type annotation